geoapps/utils/utils.py

```diff
@@ -1317,6 +1384,155 @@ def format_labels(x, y, axs, labels=None, aspect="equal", tick_format="%i", **kw
+        Only used for input entities other than BlockModel.
+
+    resolution: int, default=100
+        Grid size used to generate the iso surface.
+        Only used for input entities other than BlockModel.
+
+    Returns
+    -------
+    surfaces: list
+        List of surfaces (one per levels) defined by
+        vertices and cell indices.
+        [(vertices, cells)_level_1, ..., (vertices, cells)_level_n]
+    """
+    if getattr(entity, "vertices", None) is not None:
+        locations = entity.vertices
+    elif getattr(entity, "centroids", None) is not None:
+        locations = entity.centroids
+    else:
+        print("Input 'entity' must have 'vertices' or centroids'")
+        return None
+
+    if isinstance(entity, BlockModel):
+        values = values.reshape(
+            (entity.shape[2], entity.shape[0], entity.shape[1]), order="F"
+        ).transpose((1, 2, 0))
+        grid = [
+            entity.u_cell_delimiters,
+            entity.v_cell_delimiters,
+            entity.z_cell_delimiters,
+        ]
+
+    else:
+        grid = []
+        for ii in range(3):
+            grid += [
+                np.arange(
+                    locations[:, ii].min(),
+                    locations[:, ii].max() + resolution,
+                    resolution,
+                )
+            ]
+
+        y, x, z = np.meshgrid(grid[1], grid[0], grid[2])
+        values = weighted_average(
+            locations,
+            np.c_[x.flatten(), y.flatten(), z.flatten()],
+            [values],
+            threshold=1e-1,
+            n=8,
+            max_distance=max_distance,
+        )
+        values = values[0].reshape(x.shape)
+
+    surfaces = []
+    for level in levels:
+        try:
+            verts, faces, _, _ = marching_cubes(values, level=level)
+
+            # Remove all vertices and cells with nan
+            nan_verts = np.any(np.isnan(verts), axis=1)
+            rem_cells = np.any(nan_verts[faces], axis=1)
+
+            active = np.arange(nan_verts.shape[0])
+            active[nan_verts] = nan_verts.shape[0]
+            _, inv_map = np.unique(active, return_inverse=True)
+
+            verts = verts[nan_verts == False, :]
+            faces = faces[rem_cells == False, :]
+            faces = inv_map[faces].astype("uint32")
+
+            vertices = []
+            for ii in range(3):
+                F = interp1d(
+                    np.arange(grid[ii].shape[0]), grid[ii], fill_value="extrapolate"
+                )
+                vertices += [F(verts[:, ii])]
+
+            if isinstance(entity, BlockModel):
+                vertices = rotate_xy(np.vstack(vertices).T, [0, 0, 0], entity.rotation)
+                vertices[:, 0] += entity.origin["x"]
+                vertices[:, 1] += entity.origin["y"]
+                vertices[:, 2] += entity.origin["z"]
+
+            else:
+                vertices = np.vstack(vertices).T
+        except RuntimeError:
+            vertices, faces = [], []
```

Comment:
**suggestion:** should log the detail of the exception?
we do not really expect one, do we?

Reply:
if vertices is empty yes

Reply:
in that case, it might be clearer to test for empty `vertices` instead of letting the exception happen